tests/ImageSharp.Tests/MemoryAllocatorValidator.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Diagnostics;
+using SixLabors.ImageSharp.Diagnostics;
+using Xunit;
+
+namespace SixLabors.ImageSharp.Tests
+{
+    public static class MemoryAllocatorValidator
+    {
+        public static IDisposable MonitorAllocations(int max = 0)
```

Comment:
```suggestion
        public static TestMemoryAllocatorDisposable MonitorAllocations(int max = 0)
```
No point for `TestMemoryAllocatorDisposable` being a struct, if we box it into `IDisposable` in the end.